File: LeetCode/Easy/455-Assign_Cookies/455-AssignCookies.py
class Solution:
    def findContentChildren(self, g: list[int], s: list[int]) -> int:
        n = len(g)
        m = len(s)
        g.sort()
        s.sort()

        j = 0
        count = 0

        for i in range(n):
            while j < m and g[i] > s[j]:
                j += 1

            if j < m:
                count += 1
                j += 1
            else:
                break

        return count


a = Solution()
print(a.findContentChildren(g=[1, 2, 3], s=[1, 1]))
print(a.findContentChildren(g=[1, 2], s=[1, 2, 3]))

File: LeetCode/Easy/455-Assign_Cookies/455-AssignCookies.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution
{
public:
    int findContentChildren(vector<int> &g, vector<int> &s)
    {
        int n = g.size();
        int m = s.size();

        sort(g.begin(), g.end());
        sort(s.begin(), s.end());

        int j = 0;
        int count = 0;

        for (int i = 0; i < n; i++)
        {
            while (j < m && g[i] > s[j])
                j++;

            if (j < m)
            {
                count++;
                j++;
            }
            else
                break;
        }

        return count;
    }
};